core/geometry/src/curve/CurveChainWithDistanceIndex.ts

```diff
@@ -377,19 +379,23 @@ export class CurveChainWithDistanceIndex extends CurvePrimitive {
     }
     return true;
   }
-  /** Return the start point of `this` curve. */
+  /**
+   * Return the start point of the chain.
+   * * If the chain is empty, a zero point is returned.
+   */
   public override startPoint(result?: Point3d): Point3d {
     const c = this._path.cyclicCurvePrimitive(0);
-    if (c)
-      return c.startPoint(result);
-    return Point3d.createZero(result);
+    assert(c !== undefined, "expect non-empty chain. Fix caller if this fails.");
+    return c?.startPoint(result) ?? Point3d.createZero(result);
```

Comment:
Doesn't the assert mean that `c` cannot be undefined?

```suggestion
    return c.startPoint(result);
```

Reply:
This is an example of a weak assert, where we get the benefit of the type guard, but follow up with failure logic because we aren't 100% sure the tested condition will always hold. If such an assert fires during development, we will know to fix our callers; in a released build the assert does nothing, and the failure logic is there to save the day. For this method, cursed with an unfortunate return type, returning the zero point is awkward and potentially even just as problematic as the caller invoking this method (or the ones that follow) on an empty chain, so we want to fix these callers when we discover them.